graveler/committed/tree/diff.go

```diff
@@ -0,0 +1,191 @@
+package tree
+
+import (
+	"bytes"
+
+	"github.com/treeverse/lakefs/graveler"
+)
+
+type IteratorValue struct {
+	record *graveler.ValueRecord
+	part   *Part
+}
+
+func (iv *IteratorValue) asDiff(typ graveler.DiffType) *graveler.Diff {
+	return &graveler.Diff{
+		Type:  typ,
+		Key:   iv.record.Key,
+		Value: iv.record.Value,
+	}
+}
+
+type diffIterator struct {
+	left       Iterator
+	right      Iterator
+	leftVal    *IteratorValue
+	rightVal   *IteratorValue
+	currentVal *graveler.Diff
+	err        error
+	state      diffIteratorState
+}
+
+type diffIteratorState int
+
+const (
+	beforeInit diffIteratorState = iota
+	afterInit
+	done
+	closed
```

Comment:
1. need to have in different scope than the rest
2. need to have the same type as prefix - ex: diffIteratorStateDone